nengo_spa/algebras/hrr_algebra.py

```diff
@@ -82,6 +91,28 @@ def bind(self, a, b):
             raise ValueError("Inputs must have same length.")
         return np.fft.irfft(np.fft.rfft(a) * np.fft.rfft(b), n=n)
 
+    def fractional_bind(self, v, exponent):
+        r = np.fft.ifft(np.fft.fft(v) ** exponent)
+        if not np.allclose(r.imag, 0):
+            raise ValueError(
+                "None of the Fourier coefficients can have a "
+                "complex angle of +/- pi, otherwise it is "
+                "degenerate; self rotation is ambiguous. "
+                "Try generating the pointer using the "
+                "``nondegenerate()`` method."
+            )
+        return r.real
+
+    def make_nondegenerate(self, v):
+        v_fft = np.fft.fft(v)
+        v_fft[0] = np.abs(v_fft[0])  # zero frequency
+        d = len(v)
+        if d % 2 == 0:
+            v_fft[d // 2] = np.abs(v_fft[d // 2])  # +/- Nyquist frequency
+        r = np.fft.ifft(v_fft)
+        assert np.allclose(r.imag, 0)  # this should never happen
```

Comment:
The comment confused me ... if you are asserting the statement, it should *always* be true, shouldn't it?
Also, I'd prefer using `rfft` and `irfft` here which would avoid the issue altogether.